zkllvm/use-cases/primer.mdx

```diff
@@ -0,0 +1,222 @@
+# Primer
+
+This primer acts as an intro guide to writing complex circuits and compiling them. 
+
+## Dependency management
+
+Most circuits that cover common use cases (e.g., circuits verifying state transitions) require the use of external dependencies. 
+
+In such cases, [**using `clang` or `rustc` directly**](../getting-started/compiling-a-circuit) is discouraged. Instead, it is preferable to use a dedicated build management system:
+
+* CMake for C++
+* Cargo for Rust
+
+:::tip[`clang`]
+
+When calling `clang` directly, dependency management can also be handled via the following methods:
+
+* Via the CLI
+* Via env variables
+* Via updating the system path
+
+Using CMake is still preferable as it offers several valuable features (such as external library detection) that simplify working with complex circuits.
+
+:::
+
+:::info
+
+This primer uses `crypto3` (C++) and `arkworks` (Rust) to show how to use external libraries in a circuit. The primer can be reused for any other suitable library.
+
+:::
+
+### Using CMake
+
+Switch to the working directory:
+
+```bash
+mkdir new_circuit && cd new_circuit
+```
+
+Create a directory for CMake modules:
+
+```bash
+mkdir cmake && cd cmake
+```
+
+Add the [**`CircuitCompile.cmake` module from the zkLLVM repository**](https://github.com/NilFoundation/zkLLVM/blob/master/cmake/CircuitCompile.cmake) to the `./new_circuit/cmake` directory:
+
+```
+new_circuit
+-- cmake
+    -- CircuitCompile.cmake
+```
+
+Create the circuit:
+
+```bash
+cd .. && mkdir src
+```
+
+```bash
+cd src && touch main.cpp
+```
+
+Add circuit code that references an external library:
+
+```cpp
+#include <nil/crypto3/hash/algorithm/hash.hpp>
+#include <nil/crypto3/hash/sha2.hpp>
+
+...
+
+[[circuit]] bool circuit_func() {...}
+```
+
+Create a configuration file for CMake:
+
+```bash
+cd .. && touch CMakeLists.txt
+```
+
+Inside `CMakeLists.txt`, set the project config and include the `CircuitCompile.cmake` module:
+
+```cmake
+cmake_minimum_required(VERSION 3.2)
+
+project(hashes_circuit
+    VERSION 1.0
+    DESCRIPTION "Tutorial circuit with hashes"
+    LANGUAGES CXX)
+
+list(APPEND CMAKE_MODULE_PATH "path/to/cmake")
+
+include(CircuitCompile)
+```
+
+Add the required library to the project configuration:
+
+```cmake
+set(crypto3_DIR "path/to/crypto3_headers")
+
+find_package(crypto3 REQUIRED)
+```
+
+:::tip[Boost]
```

Comment:
I would omit this tip. Boost is an internal dependency of `crypto3`, so `crypto3::all` target is responsible for providing it. We should not care about it here unless our circuit uses Boost directly. But in this case Boost does not deserve a special mention, because it's equal to any alternative library.

Reply:
Alright, this makes sense. I have removed the tip